cmd/node-cache/main.go

```diff
@@ -29,11 +30,12 @@ import (
 
 // configParams lists the configuration options that can be provided to dns-cache
 type configParams struct {
-	localIP   string        // ip address for the local cache agent to listen for dns requests
-	localPort string        // port to listen for dns requests
-	intfName  string        // Name of the interface to be created
-	interval  time.Duration // specifies how often to run iptables rules check
-	exitChan  chan bool     // Channel to terminate background goroutines
+	localIP            string        // ip address for the local cache agent to listen for dns requests
+	localPort          string        // port to listen for dns requests
+	intfName           string        // Name of the interface to be created
+	interval           time.Duration // specifies how often to run iptables rules check
+	maxIptablesLockErr int           // Max consecutive iptables lock errors to ignore before restarting
```

Comment:
when does it make sense to restart vs just keep try?

Reply:
I was thinking about this too. Since the lock error can occur when rules are being checked, it is possible that the rules are intact, but check failed, or the rules got removed and check failed as well. It would be good to recover at some point via restart - if rules cannot be added at startup, the pod will crashloop till it is added. It will cause dns downtime till the rules get added back - this is the case whether we restart or no.

Reply:
Shouldn't we make the whole process convergent? I'm concerned about getting unlucky and ending up in the pod scheduling backoff.

Reply:
Made changes to update metrics counting setup errors rather than restarting.